---
title: "Aggregating read counts and methylated counts in a region for BS-seq data"
output:
  html_document:
    toc: true
---
***

## Inputs
CpG site summary are usually generated by functions in *Bismark* and *Bsmap*. It can also be generated by third-party tools such as [MethylDackel](https://github.com/dpryan79/MethylDackel). Take one sample *SRX116603* as an example.

```{r eval = TRUE, echo = TRUE, message = FALSE, warning = FALSE, comment = NA}
options(stringsAsFactors = FALSE)
xFile = "/sibcb2/bioinformatics/JupyterLab/dnaMethylation/BSMAP/SRX116603/mCall/SRX116603_CpG.bedGraph"
Tx = read.table(file = xFile, row.names = NULL, header = FALSE, sep = "\t", skip = 1, nrows = 100)
head(Tx)
```
This is a [bedGraph](https://genome.ucsc.edu/goldenPath/help/bedgraph.html) format, in which the chromosome coordinates are zero-based, half-open.

*Bismark* coverage file has similar but different format in which chromosome coordinates are 1-based. By default, only CpG sites were included in this file.

```{r eval = TRUE, echo = TRUE, message = FALSE, warning = FALSE}
yFile = "/sibcb2/bioinformatics/JupyterLab/dnaMethylation/Bismark/SRX116603/Call/SRX116603_Dedup.bismark.cov.gz"
cat(system(paste("zcat", yFile, "|head"), intern = TRUE), sep = "\n")
```

## load data

```{r eval = TRUE, echo = TRUE, message = FALSE, warning = FALSE, comment = NA}
library("methylKit")

df = read.table(file = xFile, row.names = NULL, header = FALSE, sep = "\t", skip = 1, nrows = 100)
xRaw = new("methylRaw",data.frame(chr=df[,1], start=df[,2] + 1, end=df[,3], strand="*", 
			coverage=(df[,5]+df[,6]), numCs=df[,5],numTs=df[,6]),
        	sample.id="SRX116603", assembly="hg19", context="CpG",resolution="base")

df = read.table(file = yFile, row.names = NULL, header = FALSE, sep = "\t", skip = 0, nrows = 100)
yRaw = new("methylRaw", data.frame(chr=df[,1], start=df[,2], end=df[,3], strand="*", 
			coverage=(df[,5]+df[,6]), numCs=df[,5],numTs=df[,6]),
        	sample.id="SRX116603", assembly="hg19", context="CpG",resolution="base")
head(xRaw)
```

## Region-level summary
```{r eval = TRUE, echo = TRUE, message = FALSE, warning = FALSE, comment = NA}
gr = GRanges("chr1:10470-10472")
regionCounts(xRaw,gr,cov.bases=0,strand.aware=FALSE)
```
